thunder/dynamo/utils.py

```diff
@@ -1007,32 +982,52 @@ def default_optimizer(gm: torch.fx.GraphModule, stats: ProfileStats) -> Callable
 
-    cur_gm = remove_empty_autocast(stats.gm)
-    recompile_graph(cur_gm)
-    _, subgraph_info = _splitter(cur_gm, jit, torch.compile, _unused_sample_args=None)
+    placeholders = [n for n in stats.gm.graph.nodes if n.op == "placeholder"]
+    example_inputs_meta = [_get_example_inputs_from_placeholder(p, only_metadata=True) for p in placeholders]
+    example_inputs = get_or_create_example_inputs_from_placeholders(placeholders)
 
-    thunder_split_gm = subgraph_info.original_split_graph_module
-    split_gm = copy.deepcopy(thunder_split_gm.split_graph_module)
-    for node in split_gm.graph.nodes:
-        if not node.name.startswith("submod"):
-            continue
-        graph_module = getattr(split_gm, node.name)
-        placeholders = [n for n in graph_module.graph.nodes if n.op == "placeholder"]
-        example_inputs = [_get_example_inputs_from_placeholder(p, only_metadata=False) for p in placeholders]
-        if thunder_split_gm.is_thunder_supported_partition(node):
-            optimized_module, compiler_name = default_compile_strategy(graph_module, example_inputs)
-            # Update the node name from "submod_*" to the optimizer specifed name for more user-friendly names
-            update_node_and_submodule(split_gm, node, node.name.replace("submod", compiler_name), optimized_module)
-        else:  # For inductor
-            optimized_module = _torch_inductor_compile(graph_module, example_inputs)
-            # Update the node name from "submod_*" to "inductor_*" for more user-friendly names
-            update_node_and_submodule(split_gm, node, node.name.replace("submod", "inductor"), optimized_module)
-
-    recompile_graph(split_gm)
-    return split_gm
+    report = FXGraphReport(copy.deepcopy(gm), "gm", example_inputs_meta)
+    thunderfx_compiler = ThunderCompilerOnGraphModuleSpecification(nv_skip_cache=True)
+    torchinductor = TorchInductorSpecification()
+    torcheager = TorchEagerSpecification()
+
+    def get_timing(report, compile_fn, timer_fn):
+        try:
+            measurement = report.run_benchmark(
+                compile_fn,
+                timer_fn,
+                reset_torch_dynamo=False,
+                example_inputs=example_inputs,
+                measure_fwd_bwd_together=True,
+            )
+        except Exception as e:
+            return float("inf")
+        return sum(m.median for m in measurement if m is not None)
+
+    compiled_gm_to_measurement = []
+    compiled_gm_to_measurement.append(("thunderfx", get_timing(report, thunderfx_compiler, WallTime)))
+    compiled_gm_to_measurement.append(("inductor", get_timing(report, torchinductor, WallTime)))
+    compiled_gm_to_measurement.append(("eager", get_timing(report, torcheager, WallTime)))
+
+    sorted_compiled_gm_to_measurement = sorted(compiled_gm_to_measurement, key=itemgetter(1))
+    best_option = sorted_compiled_gm_to_measurement[0][0]
```

Comment:
TODO: sort so that the thing to return is sorted, too

Doesn't the benchmarking already produce the compiled versions of the gm for inductor and thunder? Is it hard to acquire the already compiled version of the gm?